### ISSUE TITLE
DDS message deserialization failed

### ISSUE THREAD

**Issue by redrabit** [objectcomputing/OpenDDS#943]
We recently conducted two host communication experiments on a WAN network. And also, we used to Reliable QoS and Security module.
However, when the host lost the message, they immediately received a large number of messages at once.
And also printed error message like this:
```
(29309|139897186531072) Messenger::MessageDataReaderImpl::dds_demarshal deserialization failed, dropping sample.
(29309|139897186531072) Messenger::MessageDataReaderImpl::dds_demarshal deserialization failed, dropping sample.
(29309|139897186531072) Messenger::MessageDataReaderImpl::dds_demarshal deserialization failed, dropping sample.
(29309|139897186531072) Messenger::MessageDataReaderImpl::dds_demarshal deserialization failed, dropping sample.
(29309|139897186531072) Messenger::MessageDataReaderImpl::dds_demarshal deserialization failed, dropping sample.
(29309|139897186531072) Messenger::MessageDataReaderImpl::dds_demarshal deserialization failed, dropping sample.
(29309|139897186531072) Messenger::MessageDataReaderImpl::dds_demarshal deserialization failed, dropping sample.
```

What does this mean?? and Why this error occurred?

**Comment by mitza-oci** [comment]
It means the expected data was not found in the message (for example, it was too short).  What security settings (Governance) are you using for this topic?

**Comment by redrabit** [comment]
I set <data_protection_kind> to ENCRYPT for encrypting message data.
Follow is my governance.xml file.

```
<dds xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../schema/omg_shared_ca_governance.xsd">
  <domain_access_rules>
    <!-- Domain 4 is an "open domain" -->
    <domain_rule>
      <domains>
        <id>4</id>
      </domains>
      <allow_unauthenticated_participants>true</allow_unauthenticated_participants>
      <enable_join_access_control>false</enable_join_access_control>
      <discovery_protection_kind>NONE</discovery_protection_kind>
      <liveliness_protection_kind>NONE</liveliness_protection_kind>
      <rtps_protection_kind>NONE</rtps_protection_kind>
      <topic_access_rules>
        <topic_rule>
          <topic_expression>*</topic_expression>
          <enable_discovery_protection>false</enable_discovery_protection>
          <enable_liveliness_protection>false</enable_liveliness_protection>
          <enable_read_access_control>false</enable_read_access_control>
          <enable_write_access_control>false</enable_write_access_control>
          <metadata_protection_kind>NONE</metadata_protection_kind>
          <data_protection_kind>ENCRYPT</data_protection_kind>
        </topic_rule>
      </topic_access_rules>
    </domain_rule>
  </domain_access_rules>
</dds>
```

**Comment by redrabit** [comment]
To solve this problem, we captured packets by wireshark. And we found that the subscriber requested retransmission when packet loss occurred, and that all received packets were not decrypted until the lost packet arrived.

That is, if the received packet's writerSeqNumber does not have a continuous value, all packets between the retransmission requested and the retransmission received can not be decoded.

How would we solve this problem?
